Enhance Documentation: Update `contributing.qmd`, `README.md`, and `index.qmd` (#51)
cloudnativegeo/cloud-optimized-geospatial-formats-guide: contributing.qmd

title: Get Involved
---

## Introduction

> We encourage contributions to this guide. The guide's goal on cloud-optimized geospatial formats is to provide documentation on the best practices for the current state-of-the-art cloud-optimized formats. These formats are evolving, and so will the guide.

Reviewer: ```suggestion
We encourage contributions to this guide. The guide's goal is to provide documentation on the best practices for the current state-of-the-art cloud-optimized formats. These formats are evolving, and so will the guide.
```

## Pre-requisites

If you wish to preview the site locally, install [quarto](https://quarto.org/). You will also need to be familiar with [quarto markdown](https://quarto.org/docs/authoring/markdown-basics.html).

## Communication Channels

> Discussions can occur in [GitHub Discussions](https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/discussions) and issues can be raised at [GitHub Issues](https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/issues).

Reviewer: In case contributors might not be familiar with Github, maybe add one sentence for when to create a discussion vs issue?

- Landing pages with no code should use [quarto markdown (`.qmd`)](https://quarto.org/docs/authoring/markdown-basics.html).
- Pages with executable code should be [Jupyter Notebooks (`.ipynb`)](https://ipython.org/notebook.html).

## Code of Conduct

- Be inclusive, respectful, and understanding of others' backgrounds and contexts.
- Look for and foster diverse perspectives.
- If you experience any harmful behavior, please get in touch with [Aimee](mailto:aimee@developmentseed.org) or [Alex](mailto:alex@developmentseed.org).

## Bug Reporting & Feature Requests

Before submitting a bug report or a feature request, please start a [GitHub Discussion](https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/discussions) to see if the issue has already been addressed or if it can be resolved through discussion.

### General Steps

1. Fork the repository.
2. Clone your fork locally.
3. Create a new branch for your changes.
4. Make your changes and use `quarto preview` to make sure they look good.
5. Open a pull request.

> Once the pull request is opened, and the GitHub `preview.yml` workflow runs ("Deploy PR previews"), you should have a preview available for review at `https://developmentseed.org/cloud-optimized-geospatial-formats-guide/pr-preview/pr-<YOUR-PR-NUMBER-HERE>`.

Reviewer: ```suggestion
Once the pull request is opened, and the GitHub `preview.yml` workflow runs ("Deploy PR previews"), you should have a preview available for review at `https://developmentseed.org/cloud-optimized-geospatial-formats-guide/pr-preview/pr-<YOUR-PR-NUMBER-HERE>`. A bot will comment on your PR when the PR preview is ready.
```

title: Get Involved
---

## Introduction

> We encourage contributions to this guide. The guide's goal is to provide documentation on the best practices for the current state-of-the-art cloud-optimized formats. These formats are evolving, and so will the guide.

Reviewer 2: ```suggestion
We encourage contributions to this guide. The guide's goal is to document current best practices for state-of-the-art cloud-optimized formats. These best practices and formats are evolving, and so will the guide.
```

## Code of Conduct

- Be inclusive, respectful, and understanding of others' backgrounds and contexts.
- Look for and foster diverse perspectives.
- If you experience any harmful behavior, please get in touch with [Aimee](mailto:aimee@developmentseed.org) or [Alex](mailto:alex@developmentseed.org).

## Bug Reporting & Feature Requests

Before submitting a bug report or a feature request, please start a [GitHub Discussion](https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/discussions) to see if the issue has already been addressed or if it can be resolved through discussion.

### General Steps

1. Fork the repository.
2. Clone your fork locally.
3. Create a new branch for your changes.
4. Make your changes and use `quarto preview` to make sure they look good.
5. Open a pull request.

Once the pull request is opened, and the GitHub `preview.yml` workflow runs ("Deploy PR previews"), you should have a preview available for review at `https://developmentseed.org/cloud-optimized-geospatial-formats-guide/pr-preview/pr-<YOUR-PR-NUMBER-HERE>`. A bot will comment on your PR when the PR preview is ready.

### Specific Contributions

> #### 1. Adding a New Format

Reviewer 2: I've done some googling on how to capitalize subtitles and it seems guidance is inconclusive so I guess I don't care which one we go with as long as we are consistent. This almost makes me want to add a guidelines on formatting page in the future.

Reviewer: I agree that either is fine but good to be consistent. I think in my own writing I've used lower case after the first word, so we'd probably need to update those files

Author: I updated or verified `overview.qmd`, `index.qmd`, `README.md` and `contributing.qmd`

1. Create a folder with the format's name and, within that folder, an `intro.qmd`.
2. Link to the `intro.qmd` page in the `index.qmd` (the **Welcome** page) file and `_quarto.yml` table of contents.

#### 2. Modify or Add to an Existing Format

Feel free to modify or add to existing content if you think it could be improved.

#### 3. Adding a Cookbook

Cookbooks should address common questions and present solutions for cloud-optimized access and visualization. To create a cookbook, either add a notebook directly to this repository in the cookbooks directory OR use an external link and add it to cookbooks/index.qmd.

#### 4. (Optional) Update Slides

If you have made substantive changes, consider updating the [Overview Slides](./overview.qmd). These slides are generated using [Quarto and Reveal.js](https://quarto.org/docs/presentations/revealjs/) so can be updated with markdown syntax.

#### 5. Add Yourself to the List of Authors

Add yourself to the list of authors on the [Welcome](./index.qmd#authors) page.

#### 6. Final Steps Before Merging

Once your PR is approved and all checks have passed, a project maintainer will merge your changes into the main repository.

## Licensing

> Contributions to this project are accepted under the Creative Commons Attribution 4.0 International license.

Reviewer 3: TODO: figure out if people are assigning copyright to DevSeed or Radiant (so that changes can be made later without needing author permission - e.g. upgrading to a newer CC version later)

Author: Opened an issue: https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/issues/52

Reviewer 3: not a blocker for this PR, but is a blocker to accepting PRs from outside authors.
